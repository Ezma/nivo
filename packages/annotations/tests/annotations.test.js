/*
 * This file is part of the nivo project.
 *
 * Copyright 2016-present, RaphaÃ«l Benitte.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
import React from 'react'
import renderer from 'react-test-renderer'
import { linearScale } from '@nivo/scales'
import Annotation from '../src/annotations'

const xScale = linearScale({ axis: 'x' }, { x: { min: 0, max: 10 } }, 100, 100)
const yScale = linearScale({ axis: 'y' }, { y: { min: 0, max: 10 } }, 100, 100)

const annotations = [
    {
        type: 'label',
        x: 2,
        y: 3.6,
        dx: 30,
        dy: 30,
        note: {
            label: 'This is a noteworthy datapoint',
            orientation: 'leftRight',
        },
        connector: {
            type: 'elbow',
        },
    },
    {
        type: 'xyThreshold',
        x: 6.5,
        y: 2,
        dx: 30,
        dy: 30,
        note: {
            label: 'This is a vertical line',
            orientation: 'leftRight',
        },
        subject: {
            y1: 0,
            y2: 10,
        },
    },
]

it('renders a basic annotation', () => {
    const tree = renderer
        .create(<Annotation {...annotations[0]} xScale={xScale} yScale={yScale} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders a threshold line annotation', () => {
    const tree = renderer
        .create(<Annotation {...annotations[0]} xScale={xScale} yScale={yScale} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})
