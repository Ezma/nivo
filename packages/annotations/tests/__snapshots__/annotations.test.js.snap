// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic annotation 1`] = `
<g
  className="annotation label"
  transform="translate(20, 64)"
>
  <g
    className="annotation-connector"
  >
    <path
      className="connector"
      d="M0,0L30,30L30,30"
      fill="none"
      stroke="grey"
    />
  </g>
  <g
    className="annotation-subject"
  />
  <g
    className="annotation-note"
    transform="translate(30, 30)"
  >
    <g
      className="annotation-note-content"
      transform="translate(3,
          0)"
    >
      <rect
        className="annotation-note-bg"
        fill="white"
        fillOpacity="0"
        height={0}
        stroke="none"
        width={0}
        x={-0}
        y={-0}
      />
      <text
        className="annotation-note-label"
        fill="grey"
        y={0}
      >
        <tspan
          dy=".8em"
          x={0}
        >
          This is a noteworthy datapoint
        </tspan>
      </text>
    </g>
  </g>
</g>
`;

exports[`renders a threshold line annotation 1`] = `
<g
  className="annotation label"
  transform="translate(20, 64)"
>
  <g
    className="annotation-connector"
  >
    <path
      className="connector"
      d="M0,0L30,30L30,30"
      fill="none"
      stroke="grey"
    />
  </g>
  <g
    className="annotation-subject"
  />
  <g
    className="annotation-note"
    transform="translate(30, 30)"
  >
    <g
      className="annotation-note-content"
      transform="translate(3,
          0)"
    >
      <rect
        className="annotation-note-bg"
        fill="white"
        fillOpacity="0"
        height={0}
        stroke="none"
        width={0}
        x={-0}
        y={-0}
      />
      <text
        className="annotation-note-label"
        fill="grey"
        y={0}
      >
        <tspan
          dy=".8em"
          x={0}
        >
          This is a noteworthy datapoint
        </tspan>
      </text>
    </g>
  </g>
</g>
`;
